src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -176,6 +176,15 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleSwitchView(commandResult.getTargetView().trim());
             }
 
+            if (commandResult.isUndo()) {
+                logic.getModel().setState((ElisaStateManager)
```

Comment:
Is there any particular reason why there is a need for type casting here?

Reply:
The method was missing from the interface, I have added it.